internal/common/rules/ExampleMatrixRegistration.go

```diff
@@ -0,0 +1,30 @@
+package rules
+
+import (
+	"gonum.org/v1/gonum/mat"
+)
+
+func RegisterCoolRool() {
+
+	//A very contrived rule//
+
+	registerNewVariable("number_of_islands_contributing_to_common_pool", 5)
+	registerNewVariable("number_of_failed_forages", 0.5)
+	registerNewVariable("number_of_broken_agreements", 1)
+	registerNewVariable("max_severity_of_sanctions", 2)
+
+	name := string("Kinda Complicated Rule")
+	reqVar := make([]string, 4)
```

Comment:
```golang
reqVar := []string{
     "number_of_islands_contributing_to_common_pool",
     ....
}
```
should work.

These variables should also be `enum`s IMO? (in golang they're `iota`s that instead of return `int`s, but we could just have const string literals

Reply:
They should be exported as well for clients to read it. we can have a file with just these constants

Reply:
Yeah tbh enum would be a lot cleaner, I wasn't thinking that far ahead tbh, and also with the self-generated rules I wasn't sure how that works.

Reply:
ah--i just saw these being used in `registerNewVariable`. I think having them as exported constnats in a separate file (rules are bound to have shared variables at some point I think?) would be better

Reply:
as discussed we'll stick to strings for now